src/daft-recordbatch/src/ops/window.rs

```diff
@@ -127,19 +131,112 @@ impl RecordBatch {
             }
         };
 
-        self.window_agg_rows_between(
-            agg_expr,
-            &name,
-            dtype,
-            start_boundary,
-            end_boundary,
-            min_periods,
-            total_rows,
-        )
+        let source = self.get_column(agg_expr.name())?;
+        // Check if we can initialize an incremental state
+        match create_window_agg_state(source, agg_expr, total_rows) {
+            Ok(Some(agg_state)) => {
+                // Incremental state created successfully
+                self.window_agg_rows_incremental(
+                    &name,
+                    start_boundary,
+                    end_boundary,
+                    min_periods,
+                    total_rows,
+                    agg_state,
+                )
+            }
+            Ok(None) => {
+                // Otherwise, use the non-incremental implementation
+                self.window_agg_rows(
+                    agg_expr,
+                    &name,
+                    dtype,
+                    start_boundary,
+                    end_boundary,
+                    min_periods,
+                    total_rows,
+                )
+            }
+            Err(e) => {
+                // Error during setup (e.g., type incompatibility), propagate it
+                Err(e)
+            }
```

Comment:
We can use Rust's question mark syntax here instead and just match on the option: 
`match create_window_agg_state(source, agg_expr, total_rows)? {`